src/rust/multi.ninemanga/src/parser.rs

```diff
@@ -0,0 +1,299 @@
+		let html = Request::new(url.as_str(), HttpMethod::Get)
+			.header("Accept-Language", "es-ES,es;q=0.9,en;q=0.8,gl;q=0.7")
+			.header(
+				"User-Agent",
+				"Mozilla/5.0 (Windows NT 10.0; WOW64) Gecko/20100101 Firefox/75",
+			)
+			.html()
+			.expect("Failed to initialize the request");
+
+		let title = String::from(
+			html.select(".bookintro li")
+				.select("span")
+				.text()
+				.read()
+				.replace("Manga", "")
+				.trim(),
+		);
+
+		let cover = html.select(".bookface img").attr("src").read();
+
+		let author = html
+			.select(".bookintro li")
+			.select("[itemprop='author']")
+			.text()
+			.read();
+
+		let description = String::from(
+			html.select(".bookintro p")
+				.text()
+				.read()
+				.replace("Summary:", "")
+				.trim(),
+		);
+
+		let status = status_from_string(html.select(".bookintro .red").first().text().read());
+		let mut categories = Vec::new();
+		let mut nsfw = MangaContentRating::Safe;
+		let mut viewer = MangaViewer::Rtl;
+
+		let nsfw_genres = [
+			"Adult",
+			"Mature",
+			"Ecchi",
+			"Smut",
+			"adulto",
+			"Maduro",
+			"Hentai",
+			"Adulto",
+			"Adulto (18+)",
+			"Adulto (YAOI)",
+			"Adulte",
+		];
+
+		for node in html.select("[itemprop='genre'] a").array() {
+			let node = node.as_node().expect("Failed to get the node");
+			let category = node.text().read();
+
+			if nsfw_genres.contains(&category.as_str()) {
+				nsfw = MangaContentRating::Nsfw;
+			}
+
+			if category.as_str() == "Webtoon" {
+				viewer = MangaViewer::Scroll;
+			}
+
+			categories.push(category);
+		}
+
+		Ok(Manga {
+			id: id.clone(),
+			cover,
+			title,
+			author,
+			description,
+			url: format!("{}/manga/{}", self.base_url, id),
+			categories,
+			status,
+			nsfw,
+			viewer,
+			..Default::default()
+		})
+	}
+
+	// parse the chapters list present on manga details page
+	pub fn parse_chapter_list(&self, id: String) -> Result<Vec<Chapter>> {
+		let mut chapters: Vec<Chapter> = Vec::new();
+
+		let url = format!("{}/manga/{}?waring=1", self.base_url, id);
+		let html = Request::new(url.as_str(), HttpMethod::Get)
+			.header("Accept-Language", "es-ES,es;q=0.9,en;q=0.8,gl;q=0.7")
+			.header(
+				"User-Agent",
+				"Mozilla/5.0 (Windows NT 10.0; WOW64) Gecko/20100101 Firefox/75",
+			)
+			.html()
+			.expect("Failed to initialize the request");
+
+		let name = String::from(
+			html.select(".bookface img")
+				.attr("alt")
+				.read()
+				.replace("Manga", "")
+				.replace("Манга", "")
+				.trim(),
+		);
+
+		for chapter in html.select("ul.sub_vol_ul > li").array() {
+			let chapter_node = chapter.as_node().expect("Failed to get the node");
+
+			let raw_title = String::from(
+				chapter_node
+					.select("a")
+					.text()
+					.read()
+					.replace("13610", "")
+					.trim(),
+			);
+
+			let chapter_id = chapter_node
+				.select("a.chapter_list_a")
+				.attr("href")
+				.read()
+				.replace(".html", "-10-1.html");
+
+			let url = chapter_id.clone();
```

Comment:
chapter id should be the relative url, not the full url, just like manga ids

Reply:
Done.